File: order-management-frontend/src/types/order.ts
export interface Order {
    id: string;
    cliente: string;
    produto: string;
    valor: number;
    status: 'Pendente' | 'Processando' | 'Finalizado';
    dataCriacao: string;
}

export interface CreateOrderInput {
    cliente: string;
    produto: string;
    valor: number;
}

File: OrdersApi/Models/Order.cs
using System.ComponentModel.DataAnnotations;

namespace OrdersApi.Models
{
    public class Order
    {
        [Key]
        public Guid Id { get; set; }
        public string Cliente { get; set; } = default!;
        public string Produto { get; set; } = default!;
        public decimal Valor { get; set; }
        public string Status { get; set; } = "Pendente"; // Pendente, Processando, Finalizado
        public DateTime DataCriacao { get; set; } = DateTime.UtcNow;
    }
}
